awx/main/managers.py

```diff
@@ -127,11 +129,21 @@ def register(self, uuid=None, hostname=None, ip_address=None, node_type='hybrid'
                         other_inst.save(update_fields=['ip_address'])
                         logger.warning("IP address {0} conflict detected, ip address unset for host {1}.".format(ip_address, other_hostname))
 
-            # Return existing instance that matches hostname
+            # get the instance based on the hostname
             instance = self.filter(hostname=hostname)
+
+            # Check or update the existing uuid
+            if uuid is not None and uuid != UUID_DEFAULT:
+                if self.filter(uuid=uuid).exists():
+                    instance = self.filter(uuid=uuid)
+
+            # Return existing instance
             if instance.exists():
                 instance = instance.get()
```

Comment:
I think I had started to say something about this previously but got distracted.  This should probably be written a bit more defensively, since it will still get here and will error if there are multiple `Instance` objects pulled in by the queryset.  The most obvious option would be to use `.first()` instead of `.get()`, but we could also do something like with the conflicting ip address detection towards the top of this method instead.